packages/iTwinUI-react/src/core/ButtonGroup/ButtonGroup.test.tsx

```diff
@@ -79,6 +79,47 @@ it.each(['start', 'end'] as const)(
   },
 );
 
+it.each(['start', 'end'] as const)(
+  'should handle overflow when available space is smaller than one element (%s)',
+  (overflowPlacement) => {
+    const scrollWidthSpy = jest
+      .spyOn(HTMLElement.prototype, 'scrollWidth', 'get')
+      .mockReturnValue(200);
+    const offsetWidthSpy = jest
+      .spyOn(HTMLElement.prototype, 'offsetWidth', 'get')
+      .mockReturnValue(50);
+
+    const OverflowGroup = () => {
+      const buttons = [...Array(3)].map((_, index) => (
+        <IconButton key={index}>
+          <SvgPlaceholder />
+        </IconButton>
+      ));
+      return (
+        <ButtonGroup
+          overflowButton={() => (
+            <IconButton>
+              <SvgMore />
+            </IconButton>
+          )}
+          overflowPlacement={overflowPlacement}
+        >
+          {buttons}
+        </ButtonGroup>
+      );
+    };
+    const { container } = render(<OverflowGroup />);
+
+    expect(container.querySelector('.iui-button-group')).toBeTruthy();
+
+    const buttons = container.querySelectorAll('.iui-button');
+    expect(buttons).toHaveLength(1);
```

Comment:
Maybe we should also test a more complex case as mentioned in the issue? Something about the overflow not restoring after coming down to just one button (the `overflowButton`).

So maybe after this test, you could change the mock to simulate three buttons being displayed. Then you can verify if 3 buttons are displayed or 10 (9 (because `.slice(0, -1)` gives all elements except the last one) + 1 (overflow button)).

Reply:
> So maybe after this test, you could change the mock to simulate three buttons being displayed. Then you can verify if 3 buttons are displayed or 10 (9 (because .slice(0, -1) gives all elements except the last one) + 1 (overflow button)).

I just realized, I was thinking there are 10 buttons since there were 10 on the storybook's Overflow example. So in your case, I would mean: 

*"So maybe after this test, you could change the mock to simulate **2 buttons** being displayed. Then you can verify if **2 buttons** are displayed or **3** (**2** (because .slice(0, -1) gives all elements except the last one) + 1 (overflow button))."*

Reply:
I don't think it's necessary because the button group only doesn't restore itself after breaking, and my test checks that it doesn't break. Thoughts?

Reply:
That makes sense. Also, I now think that adding it might make the test more complex than needed. Also, if we add it, future contributors might be confused about why we are testing such a specific case. So I too now feel that there's no need of adding it